### PR TITLE
stabilize [test_id:4746] by extracting libvirt's pid in a different way

### PR DESCRIPTION
**Pull request by Barakmor1**
**What this PR does / why we need it**:
The test run 'ps -x' through virt-luncher in order to extract libvirtd pid and for unclear reasons sometimes 'ps -x' generate the **expected output** with exit code 1. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Instead of checking the exit code i modified the test to check if the pid extracted is in the right format and print all the information needed in case of an error.
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
replacing the command that extract libvirtd's pid  to avoid this error:
`ps -x`      ->      `pidof <LIBVIRTD_PID>`

```

### CODE REVIEW

**Comment by xpivarc** [comment]
/ok-to-test
@Barakmor1 I think this contains some commits that you don't necessarily want.

File: tests/migration_test.go
Comment (fgimenez) [inline tests/migration_test.go:661]: We should remove the `[QUARANTINE]` tag after we have checked that the fix worked and the test is actually stable, you can read about the quarantine procedure here https://github.com/kubevirt/kubevirt/blob/main/docs/quarantine.md

**Review by fgimenez — CHANGES_REQUESTED** [review]

File: tests/migration_test.go
Comment (Barakmor1) [reply tests/migration_test.go:661]: Hey Federico, 
Thanks for the note, As soon as this PR passes presubmit lanes I will return quarantine and stop holding this pr.

**Comment by Barakmor1** [comment]
retest

**Comment by Barakmor1** [comment]
retest

**Comment by Barakmor1** [comment]
Hey @fgimenez can you take a look?

File: tests/migration_test.go
Comment (iholder-redhat) [inline tests/migration_test.go:341]: This was previously above `pid = words[0]`, I think you should move it back there (if error is not nil there is no point to continue)

File: tests/migration_test.go
Comment (iholder-redhat) [inline tests/migration_test.go:345]: nit: you can also write it this way: `Expect(pid).ToNot(BeEmpty(), ...)`, a bit more declarative :)

**Review by iholder-redhat — COMMENTED** [review]
Thanks @Barakmor1! Looking good!

@fgimenez Although it's outside of the scope of this PR we should maybe dig into why `ps` exits with code 1 although it produced a valid output.

File: tests/migration_test.go
Comment (Barakmor1) [reply tests/migration_test.go:339]: done

File: tests/migration_test.go
Comment (Barakmor1) [reply tests/migration_test.go]: done

**Comment by iholder-redhat** [comment]
Thanks a lot! Looks good!
/lgtm

File: tests/migration_test.go
Comment (fgimenez) [reply tests/migration_test.go]: awesome thanks!

**Comment by jean-edouard** [comment]
Please update the PR title, it doesn't seem like you take any test out of quarantine anymore, right?
Also, can you think of a good reason not to use `pidof` here? Parsing the output of `ps` is quite horrendous, especially since we basically grep all running processes for any match of "libvirtd"!

**Comment by Barakmor1** [comment]
> Please update the PR title, it doesn't seem like you take any test out of quarantine anymore, right? Also, can you think of a good reason not to use `pidof` here? Parsing the output of `ps` is quite horrendous, especially since we basically grep all running processes for any match of "libvirtd"!



> Please update the PR title, it doesn't seem like you take any test out of quarantine anymore, right? Also, can you think of a good reason not to use `pidof` here? Parsing the output of `ps` is quite horrendous, especially since we basically grep all running processes for any match of "libvirtd"!

Thanks for the insights i will try that.

File: tests/migration_test.go
Comment (iholder-redhat) [inline tests/migration_test.go:323]: Looks muck better!

**Review by iholder-redhat — COMMENTED** [review]
Thanks!
/lgtm

**Review by fgimenez — APPROVED** [review]
/lgtm

Thanks!

**Comment by jean-edouard** [comment]
Looks good! So we're still keeping the test in quarantine? What does it mean, do we run quarantined tests and ignore their outcome, or do we just not run them?
Will the flake checker still verify that test? Either way, we need a plan for this test to get out of quarantine.

Another thing, that was already brought up above, is that the commits for this PR should probably be squashed, as commit 1 and 2 kind of cancel each other!

**Comment by iholder-redhat** [comment]
> Looks good! So we're still keeping the test in quarantine? What does it mean, do we run quarantined tests and ignore their outcome, or do we just not run them? Will the flake checker still verify that test? Either way, we need a plan for this test to get out of quarantine.
> 
> Another thing, that was already brought up above, is that the commits for this PR should probably be squashed, as commit 1 and 2 kind of cancel each other!

Hey @jean-edouard!
According to our [quarantine process](https://github.com/kubevirt/kubevirt/blob/main/docs/quarantine.md), in order to take test out of quarantine we need to keep it running in periodics (which run quarantined tests) and expect no (or almost no) failures during 14 days period. When this is done, test can be taken out of quarantine.

**Comment by fgimenez** [comment]
> Looks good! So we're still keeping the test in quarantine? What does it mean, do we run quarantined tests and ignore their outcome, or do we just not run them?

@jean-edouard sure, once the PR is merged we check for 14 days how it behaves in periodics from master (all the quarantined tests are executed here), if there are no more errors we propose a PR to remove the quarantine flag and execute again in presubmits. You can read about the quarantine process here https://github.com/kubevirt/kubevirt/blob/main/docs/quarantine.md

**Comment by fgimenez** [comment]
As Itamar said ;)

**Pull request merged**
